artwork/Modules/CostCenter/Services/CostCenterService.php

```diff
@@ -2,11 +2,26 @@
 
 namespace Artwork\Modules\CostCenter\Services;
 
+use Artwork\Modules\CostCenter\Models\CostCenter;
 use Artwork\Modules\CostCenter\Repositories\CostCenterRepository;
 
 readonly class CostCenterService
 {
     public function __construct(private CostCenterRepository $costCenterRepository)
     {
     }
+
+    public function findOrCreateCostCenter(string $name): int
+    {
+        $createdOrFoundedCostCenter = $this->costCenterRepository->findCostCenterByName($name);
+
+        if ($createdOrFoundedCostCenter === null) {
+            $costCenter = new CostCenter();
```

Comment:
MUST:
Damit diese Funktion testbar wird muss das "new CostCenter()" in eine Funktion im Service ausgelagert werden. Ein mit new instanziiertes Objekt lässt sich nicht mocken. Normalerweise löst man sowas über das Factory Design Pattern, das ist hier aber zuviel des Guten. (Den Fehler habe ich leider auch schon öfters gemacht und das müssen wir alles anpassen in Zukunft :/)

`
public function createNewCostCenter(): CostCenter {
    return new CostCenter();
}
`

Wenn diese eine Funktion dann nicht getestet werden kann ist das nicht so schlimm. Die Logik in der findOrCreateCostCenter-Funktion ist wichtig zu testen.

Weiterer Benefit ist das wenn man n neues Model braucht, man an anderen Stellen dazu einfach den Service verwenden kann, wodurch diese Sachen dann auch testbar und konsistent bleiben.